e2e_tests/tests/fixtures/keras_no_op/model_def.py

```diff
@@ -0,0 +1,81 @@
+import random
+
+import numpy as np
+import tensorflow as tf
+from packaging import version
+
+from determined.keras import TFKerasTrial, TFKerasTrialContext
+
+
+class RandomMetric(tf.keras.metrics.Metric):
+    def update_state(self, *args, **kwargs):
+        return None
+
+    def result(self):
+        def my_func(x):
+            return random.random()
+
+        return tf.compat.v1.py_func(my_func, [tf.ones([1], dtype=tf.float64)], tf.float64)
+
+
+class NumPyRandomMetric(tf.keras.metrics.Metric):
+    def update_state(self, *args, **kwargs):
+        return None
+
+    def result(self):
+        def my_func(x):
+            return np.random.random()
+
+        return tf.compat.v1.py_func(my_func, [tf.ones([1], dtype=tf.float64)], tf.float64)
+
+
+class TensorFlowRandomMetric(tf.keras.metrics.Metric):
+    def update_state(*args, **kargs):
+        pass
+
+    def result(self):
+        def my_func(x):
+            if version.parse(tf.__version__) >= version.parse("2.0.0"):
+                return tf.random.get_global_generator().uniform
+            else:
+                return 0.0
+
+        return tf.compat.v1.py_func(my_func, [tf.ones([1], dtype=tf.float64)], tf.float64)
+
+
+class NoopKerasTrial(TFKerasTrial):
+    def __init__(self, context: TFKerasTrialContext):
+        self.context = context
+
+    def build_model(self):
+        model = tf.keras.Sequential(
+            [
+                tf.keras.layers.Dense(
+                    8,
+                    input_shape=(
+                        8,
+                        8,
+                    ),
+                )
+            ]
+        )
+        model = self.context.wrap_model(model)
+        model.compile(
+            loss=tf.keras.losses.MeanSquaredError(),
+            metrics=[
```

Comment:
q: are the `result()` of those metrics called once per batch?

Reply:
Yes, I believe so.